velero-plugins/pod/restore.go

```diff
@@ -114,26 +139,37 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	podHasRestoreHooks := false
 	p.Log.Info("[pod-restore] checking if pod has restore hooks")
 	if input.Restore.Spec.Hooks.Resources != nil {
-		podHasRestoreHooks, err = p.podHasRestoreHooks(pod, input.Restore.Spec.Hooks.Resources)
+		podHasRestoreHooks, err = PodHasRestoreHooks(pod, input.Restore, p.Log)
 		if err != nil {
 			p.Log.Errorf("[pod-restore] checking if pod has restore hooks failed, got error: %s", err.Error())
 			return nil, err
 		}
 	}
 
+	p.Log.Info("[pod-restore] checking if pod has volumes to back up")
+	podHasVolumesToBackUp := PodHasVolumesToBackUp(pod)
+
 	// Check if pod has owner Refs and defaultVolumesToRestic flag as false/nil
-	if (len(ownerRefs) > 0 && pod.Annotations["backup.velero.io/backup-volumes-excludes"] == "" && (defaultVolumesToFsBackup == nil || !*defaultVolumesToFsBackup)) && !podHasRestoreHooks {
-		p.Log.Infof("[pod-restore] skipping restore of pod %s, has owner references, no restic backup, and no restore hooks", pod.Name)
+	if len(ownerRefs) > 0 && !podHasVolumesToBackUp && !podHasRestoreHooks {
+		p.Log.Infof("[pod-restore] skipping restore of pod %s, has owner references, no volumes to back up, and no restore hooks", pod.Name)
 		return velero.NewRestoreItemActionExecuteOutput(input.Item).WithoutRestore(), nil
 	}
 
 	// If pod has both "deployment" and "deploymentconfig" labels, it belongs to a DeploymentConfig
-	// If defaultVolumesToRestic, remove these labels so that the DC won't immediately delete the
-	// pod on restore, and add disconnected-from-dc label with restore name for post-restore cleanup
+	// For pre-OADP1.3 restores, if defaultVolumesToRestic, remove these labels so that the DC won't
+	// immediately delete the pod on restore, and add disconnected-from-dc label with restore name for
+	// post-restore cleanup
+	// as of OADP1.3, base this on the presence of any volumes to back up or restore hooks
+	disconnectIfDC := false
+	if pod.Annotations != nil && len(pod.Annotations[common.DCIncludesDMFix]) > 0 {
+		disconnectIfDC = podHasRestoreHooks || podHasVolumesToBackUp
+	} else {
+		disconnectIfDC = defaultVolumesToFsBackup != nil && *defaultVolumesToFsBackup
```

Comment:
From reading IIUC **if** is for <1.3 and **else** is for 1.3+. Correct?

Reply:
Yes -- more specifically, if *backup* was from before 1.3, else...

Reason for this is that we need to change behavior here around DeploymentConfigs to do the same thing for 1.3 datamover as we previously did for restic, but some of the required information can only be determined at backup time (which is fine, since we didn't have upstream datamover in 1.2) -- but at the same time, when restoring 1.2 backups, we need to use the old restic-specific check.

I'll add a comment to make this clearer.